frontend/amundsen_application/static/js/pages/LineagePage/index.tsx

```diff
@@ -0,0 +1,217 @@
+    getTableLineageDispatch(this.key, 5);
+    this.didComponentMount = true;
+  }
+
+  componentDidUpdate() {
+    const { getTableLineageDispatch } = this.props;
+    const newKey = this.getTableKey();
+    if (this.key !== newKey) {
+      this.key = newKey;
+      getTableLineageDispatch(this.key, 5);
+    }
+  }
+
+  private didComponentMount: boolean = false;
+
+  private key: string;
+
+  getDisplayName() {
+    const { match } = this.props;
+    const { params } = match;
+
+    return `Lineage Information | ${params.schema}.${params.table}`;
+  }
+
+  // ToDo: Should be moved to a common place
+  getTableKey() {
+    /*
+    This 'key' is the `table_uri` format described in metadataservice. Because it contains the '/' character,
+    we can't pass it as a single URL parameter without encodeURIComponent which makes ugly URLs.
+    DO NOT CHANGE
+    */
+    const { match } = this.props;
+    const { params } = match;
+
+    return `${params.database}://${params.cluster}.${params.schema}/${params.table}`;
+  }
+
+  render() {
+    const { match, lineageTree, statusCode, isLoading } = this.props;
+    const { params } = match;
+    let innerContent;
+    // We want to avoid rendering the previous table's metadata before new data is fetched in componentDidMount
+    if (isLoading || !this.didComponentMount) {
+      innerContent = (
+        <div className="pulse-lineage">
+          <div className="pulse-lineage-media is-pulse-animated" />
+        </div>
+      );
+    } else if (statusCode === SERVER_ERROR_CODE) {
+      innerContent = <ErrorMessage />;
+    } else {
+      const rootNodeData = {
+        parent: '',
+        source: params.database,
+        key: this.getTableKey(),
+        badges: [],
+        usage: 0,
+        database: params.database,
+        schema: params.schema,
+        name: params.table,
+        cluster: params.cluster,
+        level: 0,
+      };
+      // This will be needed to generate the lineage graph.
+      lineageTree.upstream_entities.push(rootNodeData);
+      lineageTree.downstream_entities.push(rootNodeData);
+
+      innerContent = (
+        <div className="resource-detail-layout lineage-page">
+          <header className="resource-header">
+            <div className="header-section">
+              <span
+                className={
+                  'icon icon-header ' +
+                  getSourceIconClass(rootNodeData.database, ResourceType.table)
+                }
+              />
+            </div>
+            <div className="header-section header-title">
+              <h1
+                className="header-title-text truncated"
+                title={`${rootNodeData.schema}.${rootNodeData.name}`}
+              >
+                {rootNodeData.schema}.{rootNodeData.name}
+                <span className="text-secondary lineage-graph-label">
+                  Lineage Graph
+                </span>
+              </h1>
+              <div className="text-body-w2">
```

Comment:
This is kind or OK, but still better to use a .lineage-graph-back-link class and extend its styles to text-body-w2